src/custom/state/claim/hooks/index.ts

```diff
@@ -385,6 +385,61 @@ export function useUsdcPrice(): string | null {
 
 type VCowPriceFnNames = 'nativeTokenPrice' | 'gnoPrice' | 'usdcPrice'
 
+/**
+ * Hook that calls specific method on the vCow contract defined on the VCowMethods type
+ * and returns the value in the form of JSBI.BigInt
+ *
+ * @param method method that will be called on the vCowContract
+ */
+type VCowMethods = 'swappableBalanceOf' | 'balanceOf'
```

Comment:
Since you are very likely to call both at the same time, I suggest using https://github.com/gnosis/cowswap/blob/b85ca42876152e7878d339e4c48e92e4cdca1120/src/state/multicall/hooks.ts#L158

It's more efficient as it does a single RPC call for both queries rather than 2 individual calls.

Reply:
Good suggestion

Reply:
But I still need to call it 2 times for two different methods on the contract no?
The `methodName` param is just a string not an array of strings.
My understanding is that this hook can call one method on the contract but with multiple input values.
So I don't think this improves anything in this case.

Reply:
As the comment above the method says `formats many calls to a **single function** on a single contract, with the function name and inputs specified`

Reply:
I've used `useSingleCallResult` that calls `useSingleContractMultipleData` under the hood

Reply:
Hmmm ok, maybe that doesn't exist as I though.
In fact, at some point I did implement something like that: https://github.com/gnosis/cowswap/blob/4920289ba3034204b64f9e056d0b5b3f3a9207a5/src/custom/state/multicall/utils.ts#L213

The main difference is that it was an async function rather than a hook. We decided to go a different route so that has been removed in the end.

Anyway, I think you are in the right direction with `useSingleCallResult`.
And on the latest merge, the multicall stuff has been migrated into it's own package https://github.com/Uniswap/redux-multicall